sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/TableServiceClientBuilder.java

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.data.tables;
+
+import com.azure.core.annotation.ServiceClientBuilder;
+
+/**
+ * builds the table service clients
+ */
+@ServiceClientBuilder(serviceClients = {TableServiceClient.class, TableServiceAsyncClient.class})
+public class TableServiceClientBuilder {
+
+    private String connectionString;
+
+    /**
+     * Sets the connection string to help build the client
+     *
+     * @param connectionString the connection string to the storage account
+     * @return the TableServiceClientBuilder
+     */
+    public TableServiceClientBuilder connectionString(String connectionString) {
```

Comment:
The builder should also support either `TokenCredential` or `AzureKeyCredential`.